pkg/provider/provider.go

```diff
@@ -179,6 +181,20 @@ func (p *Provider) buildServicesAndRoutersForService(t *topology.Topology, cfg *
 			cfg.TCP.Services[key] = p.buildTCPServiceFromService(t, svc, svcPort.TargetPort.IntVal)
 			cfg.TCP.Routers[key] = buildTCPRouter(rule, entrypoint, key)
 		}
+
+	case k8s.ServiceTypeUDP:
```

Comment:
What happens if a service has both TCP and UDP ports?

Or the default type is TCP, and a user adds a UDP port?

Reply:
To be consistent with the traffic type definition, if the service `traffic-type` is `http` or `tcp` the `UDP` ports are not available through Maesh. If the service traffic type is `udp` only the UDP ports will be available.

In my humble opinion, this behaviour seems a bit weird because sometimes it doesn't make sense for a service to have only one traffic type (eg. a DNS service uses TCP and UDP). This limitation already exists if your service uses TCP for a custom app protocol and HTTP (you'll have to set the `traffic-type` to `tcp` and cannot apply HTTP middlewares).

For now, the user has to create two services to bypass this limitation but this is not acceptable from my point of view. Kubernetes allows a service to have ports with a different protocol so we should support that out of the box.

Maybe the `traffic-type` annotation should only be used to specify the traffic type for a given port to apply specific middlewares and so on (I'll create the issue if you're ok with that).

For UDP, maybe I should have made UDP port available by default but this behaviour seems not consistent to me with the current `traffic-type` behaviour.

WDYT?

Reply:
I'm also feeling bad about this limitation. We are actually facing the same situation with middlewares. It would be great to all user to specify `traffic-type` as well as middlewares for a specific port. I'm thinking about something like this: 
```
// Applies on every ports
maesh.containo.us/traffic-type: "tcp"
// Applies only on the port named "api"
maesh.containo.us/api/traffic-type: "http"
maesh.containo.us/api/circuit-breaker-expression: "..."
```

Reply:
I'm also thinking about something like that. I'll try to write a proposal on that subject.